src/ape/api/config.py

```diff
@@ -1,7 +1,11 @@
 from enum import Enum
 from typing import Any, Dict, Optional, TypeVar
 
-from pydantic import BaseModel, BaseSettings
+try:
```

Comment:
we should create a `_compat.py` file to put all the error-handling pydantic imports there and import from there.

Reply:
this will also help isolate the `type: ignore` comments we are going to have to include.